### ISSUE TITLE
Code quality

### ISSUE THREAD

**Issue by archshift** [PCSX2/pcsx2#69]
Is there any enforcement of a style guide for this project? 
- Parts of code are all over the place. 
- There are tons of random extern and static members and functions that belong to no class. 
- You redefine `BOOL` for apparently no reason (?), as well as `TRUE` and `FALSE` (?). 
- C code is littered among supposedly C++ files, and from what I've seen the C standard library is used more often than the C++ counterpart. 
  - Tons of C strings all over the place as opposed to std::string
  - Where C strings aren't used, wxStrings are, which will make it very, very difficult should you ever decide to move away from Wx.
  - Basically no use of C++11
- Main directory is cluttered with MSVC solution files and build scripts (also, do you really still need to support VS 2010?).
- You have no `CONTRIBUTING.md` (!!).
- [Incorrect alignment all over the place](https://github.com/PCSX2/pcsx2/blob/master/plugins/spu2-x/src/Config.h)
